File: rb/features.rb
# YoutubeVideo SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module YoutubeVideoFeatures
  def self.make_feature(name)
    case name
    when "base"
      YoutubeVideoBaseFeature.new
    when "test"
      YoutubeVideoTestFeature.new
    else
      YoutubeVideoBaseFeature.new
    end
  end
end

File: py/features.py
# YoutubeVideo SDK feature factory

from feature.base_feature import YoutubeVideoBaseFeature
from feature.test_feature import YoutubeVideoTestFeature


def _make_feature(name):
    features = {
        "base": lambda: YoutubeVideoBaseFeature(),
        "test": lambda: YoutubeVideoTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
